tests/tape/test_tape.py

```diff
@@ -234,23 +234,6 @@ def test_state_preparation(self):
         assert tape._prep == [A]
         assert tape.get_parameters() == params
 
-    def test_state_preparation_error(self):
-        """Test that an exception is raised if a state preparation comes
-        after a quantum operation"""
-        with pytest.raises(ValueError, match="must occur prior to any quantum"):
-            with QuantumTape() as tape:
-                B = qml.PauliX(wires=0)
-                qml.BasisState(np.array([0, 1]), wires=[0, 1])
```

Comment:
This is a change in behaviour, no? What if I use `QubitStatePrep` after a gate, won't it reset the state vector and cause unexpected results?